tests/test_models/test_tbats.py

```diff
@@ -203,6 +232,23 @@ def test_decompose_forecast_output_format(small_periodic_ts, estimator):
     assert components.shape[0] == horizon
 
 
+@pytest.mark.parametrize(
+    "estimator",
+    (
+        BATSModel,
+        TBATSModel,
+    ),
+)
+def test_decompose_predict_output_format(small_periodic_ts, estimator):
+    model = estimator()
+    model.fit(small_periodic_ts)
+
+    target = small_periodic_ts[:, "segment_1", "target"].values
+    components = model._models["segment_1"]._decompose_predict()
+    assert isinstance(components, np.ndarray)
+    assert components.shape[0] == target.shape[0]
```

Comment:
This is true only if we decompose on full train, try to decompose on the part of the train

Reply:
This is always true due to specific of `_decompose_predict`. I've added tests for part of train.